astropy/wcs/wcsapi/high_level_api.py

```diff
@@ -147,7 +147,12 @@ def world_to_pixel(self, *world_objects):
 
         if unique_match:
 
-            for key, (klass, args, kwargs) in classes.items():
+            for key, (klass, args, kwargs, *rest) in classes.items():
```

Comment:
This means there is no validation that this tuple is the right length? Is that deliberate and if so why?

(Also this probably isn't the best place for that, we should write something that tests your compliance with APE 14 :thinking: :rabbit: :hole: )

Reply:
we can validate this in the if statement below though, to make sure rest has length 0 or 1?

Reply:
Added some validation